usd/src/sdf_parser/Collision.cc

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "Collision.hh"
+
+#include <string>
+
+#include <ignition/math/Pose3.hh>
+
+// TODO(adlarkin) this is to remove deprecated "warnings" in usd, these warnings
+// are reported using #pragma message so normal diagnostic flags cannot remove
+// them. This workaround requires this block to be used whenever usd is
+// included.
+#pragma push_macro ("__DEPRECATED")
+#undef __DEPRECATED
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/usd/stage.h>
+#include <pxr/usd/usdGeom/xform.h>
+#include <pxr/usd/usdPhysics/collisionAPI.h>
+#pragma pop_macro ("__DEPRECATED")
+
+#include "sdf/Collision.hh"
+#include "../UsdUtils.hh"
+#include "Geometry.hh"
+
+namespace sdf
+{
+// Inline bracke to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+//
+namespace usd
+{
+  UsdErrors ParseSdfCollision(const sdf::Collision &_collision,
+      pxr::UsdStageRefPtr &_stage, const std::string &_path)
+  {
+    UsdErrors errors;
+    const pxr::SdfPath sdfCollisionPath(_path);
+    auto usdCollisionXform = pxr::UsdGeomXform::Define(
+      _stage, sdfCollisionPath);
+    if (!usdCollisionXform)
+    {
+      errors.push_back(UsdError(sdf::usd::UsdErrorCode::FAILED_USD_DEFINITION,
+        "Not able to define a Geom Xform at path [" + _path + "]"));
+      return errors;
+    }
+
+    auto collisionPrim = _stage->GetPrimAtPath(pxr::SdfPath(sdfCollisionPath));
+    collisionPrim.CreateAttribute(pxr::TfToken("purpose"),
+        pxr::SdfValueTypeNames->Token, false).Set(pxr::TfToken("guide"));
```

Comment:
Can you add a comment that explains why these tokens are needed?

Reply:
https://github.com/ignitionrobotics/sdformat/pull/925/commits/948f3e07a6216832e4fa5c85ce004ca8feca65d2